### PR TITLE
Support test coverage with nyc and mocha

### PR DESCRIPTION
**Pull request by YongseopKim**
This commit supports test coverage with nyc and mocha.

ONE-vscode-DCO-1.0-Signed-off-by: Yongseop Kim <yons.kim@samsung.com>

### CODE REVIEW

**Comment by YongseopKim** [comment]
We have to consider things
1. Is the test running on CI?
    - because of onecc
    - it's necessary to filter tests
2. Is the test running for coverage?
    - necessary to run on vscode=1.48.0 (Note. two tests are failed on the version)
    - and it could be run with mocha reporter=mocha-xunit-reporter

**Comment by YongseopKim** [comment]
How to run?
```console
$ npm run test coverage
...

$ ls mocha_result.xml
mocha_result.xml

$ ls coverage/cobertura-coverage.xml
coverage/cobertura-coverage.xml
```

File: package.json [hunk 1]
```diff
@@ -370,6 +370,34 @@
...
+    ],
+    "instrument": true,
+    "reporter": [
+      "cobertura",
+      "lcov",
+      "html",
+      "text",
+      "text-summary"
+    ]
+  },
```
Comment (hyunsik-yoon) [inline package.json:400]: Could you share any reference about this setting?

Comment (YongseopKim) [reply package.json:400]: This is for config of nyc
- https://github.com/istanbuljs/nyc#common-configuration-options

**Comment by YongseopKim** [comment]
@llFreetimell I got this. What can I do to resolve this?

> Additional manual verification is needed
> type-fest@0.8.1 : (MIT OR CC0-1.0)

**Comment by llFreetimell** [comment]
Please add following codes to `infra/license/package-judgment.json` :D

```json
    "type-fest@0.8.1": {
      "licenses": "(MIT OR CC0-1.0)",
      "repository": "https://github.com/sindresorhus/type-fest",
      "permitted": "yes"
    },
```

**Comment by YongseopKim** [comment]
PTAL

File: src/Tests/runTest.ts [hunk 2]
```diff
@@ -45,25 +45,33 @@ import {runTests} from '@vscode/test-electron';
...
-    // NOTE: Fix the version to 1.64.0 to avoid some errors by the latest version
-    // TODO: Remove the limitation of the version
     await runTests({
-      version: '1.64.0',
+      version: codeVersion,
       extensionDevelopmentPath,
       extensionTestsPath: extensionTestsPath,
-      launchArgs: [testWorkspace]
+      launchArgs: [testWorkspace],
+      extensionTestsEnv: {isCoverage: String(isCoverage), isCiTest: String(isCiTest)}
```
Comment (llFreetimell) [inline src/Tests/runTest.ts:71]: Maybe?

```suggestion
      extensionTestsEnv: {isCoverage: string(isCoverage), isCiTest: string(isCiTest)}
```

Comment (YongseopKim) [reply src/Tests/runTest.ts]: I misunderstand. the `String` is a function like `toString`.

**Review by llFreetimell — DISMISSED** [review]

File: src/Tests/index.ts [hunk 3]
```diff
@@ -62,9 +64,19 @@ export function run(): Promise<void> {
...
+  //
+  // TODO: Enable to get string to filter from package.json
   const testFilter = '';
-  const mocha = new Mocha({fgrep: testFilter, ui: 'tdd', color: true});
+  let mocha = new Mocha({ui: 'tdd', color: true, fgrep: testFilter});
+  const isCoverage: string|undefined = env.isCoverage;
+  if (isCoverage !== undefined && isCoverage === 'true') {
+    mocha.reporter('mocha-xunit-reporter', {output: 'mocha_result.xml'});
+  }
+  const isCiTeste: string|undefined = env.isCiTest;
```
Comment (hyunsik-yoon) [inline src/Tests/index.ts:75]: Maybe?

```suggestion
  const isCiTest: string|undefined = env.isCiTest;
```

Comment (YongseopKim) [reply src/Tests/index.ts:75]: T-T you're right! Thanks!

**Comment by YongseopKim** [comment]
PTAL again

**Review by hyunsik-yoon — APPROVED** [review]

**Comment by YongseopKim** [comment]
PTAL @llFreetimell

**Review by llFreetimell — APPROVED** [review]

**Pull request merged**